### PR TITLE
🧮 Extended sequence number counter

### PR DESCRIPTION
**Pull request by danstiner**
Saw this in an RFC and I think it will make tracking NACKs easier as we won't have to deal with wrapping of two-byte sequence numbers directly.

Includes actual tests!

### CODE REVIEW

File: src/Rtp/Types.h [hunk 1]
```diff
@@ -0,0 +1,144 @@
+#pragma once
+
+#include <cstdint>
+
+typedef uint16_t rtp_sequence_num_t;
+
+/*
+* Current protocol version.
+*/
+#define RTP_VERSION    2
```
Comment (danstiner) [inline src/Rtp/Types.h:10]: These can be dropped, they all came from the RFC as sample code

File: test/test.cpp [hunk 2]
```diff
@@ -9,12 +9,5 @@
 // https://github.com/catchorg/Catch2/blob/v2.x/docs/ci-and-misc.md#precompiled-headers-pchs
 #undef TWOBLUECUBES_SINGLE_INCLUDE_CATCH_HPP_INCLUDED
 #define CATCH_CONFIG_IMPL_ONLY
-#define CATCH_CONFIG_RUNNER
+#define CATCH_CONFIG_MAIN
```
Comment (haydenmc) [inline test/test.cpp:12]: I'd like to keep our custom main so we can eventually re-route logging to the Catch2 test logging mechanisms

File: test/unit/meson.build [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+deps = [
+    dependency('libssl'),
+    dependency('libcrypto'),
+    fmt_wrap.get_variable('fmt_dep'),
+    spdlog_wrap.get_variable('spdlog_dep'),
+    catch2_wrap.get_variable('catch2_dep'),
+]
+
+exe = executable(
```
Comment (haydenmc) [inline test/unit/meson.build:27]: Not sure how I feel about having separate binaries for unit/functional/etc. test suites - does it make sense to build the individual suites as libraries that be linked into one big test executable?

Comment (haydenmc) [reply test/unit/meson.build:27]: Thinking about it more, maybe it makes sense....

File: src/Rtp/ExtendedSequenceCounter.cpp [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        return true;
+    }
+    
+    uint16_t udelta = seq - maxSeq;
+
+    /*
+    * Source is not considered stable until MIN_SEQUENTIAL packets with
+    * sequential sequence numbers have been received.
+    */
+    if (probation > 0) {
```
Comment (haydenmc) [inline src/Rtp/ExtendedSequenceCounter.cpp:33]: Braces on newline pls :>

File: src/Rtp/ExtendedSequenceCounter.cpp [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    badSeq = RTP_SEQ_MOD + 1;   /* so seq == badSeq is false */
+    cycles = 0;
+    received = 0;
+    receivedPrior = 0;
+    expectedPrior = 0;
+}
+
+std::ostream& operator<<(std::ostream & os, const ExtendedSequenceCounter& c)
+{
+    os << "ExtendedSequenceCounter { maxSeq:" << c.maxSeq << ", cycles:" << c.cycles << ", baseSeq:" << c.baseSeq << ", received:" << c.received << " }";
```
Comment (haydenmc) [inline src/Rtp/ExtendedSequenceCounter.cpp:107]: Break up big boy line pls :>

**Review by haydenmc — APPROVED** [review]
Just a couple nits, beyond that looks good to merge!

File: src/Rtp/ExtendedSequenceCounter.cpp (hunk 5 above)
Comment (danstiner) [reply src/Rtp/ExtendedSequenceCounter.cpp]: (￣･ ω･￣)

File: src/Rtp/ExtendedSequenceCounter.cpp (hunk 4 above)
Comment (danstiner) [reply src/Rtp/ExtendedSequenceCounter.cpp]: ␤

**Pull request merged**